### ISSUE TITLE
[ie8] train-2013.05.08 - frontend test failures

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#3385]
Works in other browsers, including IE9 and IE10, but not in IE8. These two
errors are shown below, and the test appears to stop executing shortly after.

https://jrgm-train0508.personatest.org/test/?filter=dialog%2Fjs%2Fmodules%2Fauthenticate%3A%20known%20secondary%20email%20declared%20in%20options%20-%20show%20password%20field

https://jrgm-train0508.personatest.org/test/?filter=dialog%2Fjs%2Fmodules%2Fauthenticate%3A%20allowUnverified%20with%20an%20unverified%20email%20declared%20in%20options%20-%20show%20password%20field

The message shown for the two test failures is:

```
1. Died on test #1: Can't move focus to the control because it is invisible, not enabled, or of a type that does 
not accept the focus. - { "name": "Error", "number": -2146826178, "description": "Can't move focus to the control
because it is invisible, not enabled, or of a type that does not accept the focus.", "message": "Can't move focus
to the control because it is invisible, not enabled, or of a type that does not accept the focus." }
```

Marking this as five star.

**Comment by jaredhirsch** [comment]
looking now

**Comment by jaredhirsch** [comment]
at [line 242 of dialog/js/modules/authenticate.js](https://github.com/mozilla/browserid/blob/dev/resources/static/dialog/js/modules/authenticate.js#L242) we give the PASSWORD_SELECTOR node focus via `dom.focus(PASSWORD_SELECTOR)`. If you set a breakpoint on that line and run the qunit tests, then ask IE8 if the node is hidden, IE8 will say that it is indeed hidden:

```
// assuming you've hit the breakpoint inside IE8 dev tools
>>x = $(PASSWORD_SELECTOR)
{...}
>>x.is(':hidden')
true
```

This explains the first error message.

Next, I'm going to git bisect recent history to see where this crept in. I suspect it's from an innocent-looking styling change.

**Comment by jaredhirsch** [comment]
This goes back at least to the previous train. Not sure if it warrants five-star status.

Looking further back through history.

**Comment by jrgm** [comment]
It's fundamentally different in this train-2013.05.08. I have passed this test in train-2013.04.26 (yes flaky). But now, we can't complete the tests at all. 

No, it may not be five-star, but needs to be worked on.

**Comment by jrgm** [comment]
So I did a git bisect with an awsbox and it points to:

```
# bad: [c204775e8f59e992d36c2eafcb23196260e54416] Update the authentication page for the differences between mobile and desktop.
```

But it's a little ambiguous since there are some test failures in an immediately prior commit (although that commit did run to completion and reported the failures). A lot of stuff landed in a short window of the merge.

So to make this clearer, I stood up two m1.small:
1. https://jrgm-3448315.personatest.org/test/ - which is '3448315 Merge pull request #3329 from mozilla/issue_3328_yamlparser_sha'
   
   This reports zero qunit failures (modulo occasional flakiness). (It does quietly note an error in the status bar about "Can't move focus to the control because it is invisible, not enabled, or of a type that does not accept the focus". but the Qunit tests report zero failures).
2. https://jrgm-c204775.personatest.org/test/ - which is 'c204775 Update the authentication page for the differences between mobile and desktop'
   
   This now fails to complete running the qunit tests, with "Died on test #1: Can't move focus to the control because it is invisible...".

So basically there was a problem lurking about focus which didn't cause the qunit tests to fail on IE8 in train-2013.04.26, but something has changed to make this much worse and tests do not complete.

**Comment by jaredhirsch** [comment]
@jrgm I force-pushed dev to bisector.personatest.org and verified that tests are passing on IE8 (though there are annoying long-running-script alerts). Care to verify as well? If not I'll just tear down bisector.

**Comment by jrgm** [comment]
Yeah, I patched train-2013.05.08 with this change and that passes fine on ie8 for me. You can tear down your bisector. 

Thanks @shane-tomlinson and @6a68!

**Comment by jaredhirsch** [comment]
woo!
